### DISCUSSION TITLE
Downloads for offline use

### DISCUSSION THREAD

**Discussion by KodonFrost** [osmandapp/OsmAnd#14036] [Q&A]
Heya! I want to use osmand for offline navigation.

Yet I have not found a way to download an entire country?
If I have to download a new region "just in time" when I first visit it, it's very much hindering the offline experience.

Silly story: Just imagine you crash with a plane in the middle of nowhere and you need a map. Now osmand tells you to just download the map. Sad.

Did I overlook it or where is the "download all" button on Android? It's extremely annoying to tap into every menu just to get a whole map...

**Comment by tommycrock** [comment]
Which country are you looking at? I usually see "regionwide maps" at the top when I click on a country, which allows you to download all the maps for the country in a click.

**Reply by KodonFrost** [reply]
For example to download Europe I need 250 clicks. That's quite some effort.

**Reply by The-MaddHatter** [reply]
I just checked Europe and click on a country you can download the standard map for that country.  Even on larger countries Spain,  France, and entire United Kingdom which is 4 countries together, you can download the entire countries as one.  Every Europe country has region wide map of the entire country.   It's very obvious as its the top of every country page.  

I will break it down for the world traveler by continent:

Now I believe there are geographically 52 countries in Europe.  THE UK's  countries are bundled as one if you just want road maps.  That means you can get all 52 countries for 49 clicks.  Thats not quite 20% of your claimed 250 clicks need.  

Norther America Continent which is 300% larger than Europe can be entirely downloaded in 6 maps or 6 clicks by your measure.  

Africa which has the most countries of any continent and is a fair bit larger than NA can be downloaded in 59 clicks by country. 

Asia minus Russia can be downloaded in 52 clicks again by country.  Including Russia which itself can be had in 1 map/click makes 53 total for Asia.

Australia and Oceania can be had in just 4 clicks (the smallest continent)

Central America 34 map/clicks again by country.

South America 15 map/clicks

Not that there is any real roads there Antarctica is just 1 map/click.

Adding this up it seems the entire sum of the land mass of planet earth can be downloaded in just 221 maps/clicks which is less than what you stated was needed for just Europe, the second smallest continent.  It seems to indicate what  Tommy Rock told you was correct and they could be downloaded by country.    So the only thing OSMand doesn't have is single maps by Continent and 1 single planet Earth map.
